Userland/Libraries/LibGL/Tex/Texture.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Vector.h>
+#include <LibGL/GL/gl.h>
+
+namespace GL
+{
+
+class Texture final
+{
+public:
+    Texture() = default;
+    ~Texture();
+
+    void upload_texture_data(GLint format, GLint internal_format, const GLvoid* pixel_data);
+
+private:
+    template<typename TCallback>
+    void swizzle(const Vector<u8>& pixels, TCallback&& callback);
```

Comment:
This won't work for any kind of `TCallback`, the definition in the cpp file is not visible to other translation units.
you can either specialise them here by forward-declaring them (`template void swizzle<SomeTypeForTCallback>(...);`), or just bring the definition into this header x)

Reply:
Going to remove the implementation of this for the moment until I flesh out the rest of the texture class data members. We can iterate on this design later in the week on Discord :P